src/System.Linq/tests/SelectTests.cs

```diff
@@ -742,6 +743,19 @@ public void Select_ResetCalledOnEnumerator_ExceptionThrown()
         }
 
         [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, ".NET core throws a NotSupportedException. See https://github.com/dotnet/corefx/pull/2959.")]
+        public void Select_ResetCalledOnEnumerator_NetFx_ThrowsNotImplementedException()
+        {
+            int[] source = new[] { 1, 2, 3, 4, 5 };
+            Func<int, int> selector = i => i + 1;
+
+            var result = source.Select(selector);
+            var enumerator = result.GetEnumerator();
+
+            Assert.Throws<NotImplementedException>(() => enumerator.Reset());
+        }
```

Comment:
Let's avoid duplicating the whole test just to differentiate NSE vs NIE.  You could have the body of the test be a non-Fact, and then have two one-line entrypoints to it that call the actual test with the expected exception type, and have those entry points be differentiated by platform. (Same comment applies if the same pattern is duplicated elsewhere.)